samza-core/src/main/java/org/apache/samza/runtime/LocalApplicationRunner.java

```diff
@@ -152,32 +159,37 @@ public ApplicationStatus status(StreamApplication streamApp) {
   }
 
   /**
-   * Create the Coordination needed by the application runner.
-   * @return {@link CoordinationUtils}
-   * @throws Exception exception
+   * Create the {@link CoordinationUtils} needed by the application runner.
+   * @return an instance of {@link CoordinationUtils}
+   * @throws Exception exceptions when creating the instance, such as {@link ClassNotFoundException} and
+   * {@link InstantiationException}
    */
-  /* package private */ CoordinationUtils getCoordinationUtils() throws Exception {
-    ApplicationConfig appConfig = new ApplicationConfig(config);
-    String clazz = appConfig.getCoordinationServiceFactoryClass();
-    if (clazz != null) {
-      CoordinationServiceFactory factory = ClassLoaderHelper.fromClassName(clazz);
+  /* package private */ CoordinationUtils createCoordinationUtils() throws Exception {
+    String jobCoordinatorFactoryClazz = config.get(JobCoordinatorConfig.JOB_COORDINATOR_FACTORY, "");
```

Comment:
Minor: xClazz name is usually used for Class<X> parameters. Maybe jobCoordinatorFactoryClassName instead?